src/gt4py/next/iterator/transforms/constant_folding.py

```diff
@@ -20,38 +20,129 @@ def visit_FunCall(self, node: ir.FunCall):
         # visit depth-first such that nested constant expressions (e.g. `(1+2)+3`) are properly folded
         new_node = self.generic_visit(node)
 
-        if (
-            isinstance(new_node.fun, ir.SymRef)
-            and new_node.fun.id in ["minimum", "maximum"]
-            and new_node.args[0] == new_node.args[1]
-        ):  # `minimum(a, a)` -> `a`
-            return new_node.args[0]
-
-        if (
-            isinstance(new_node.fun, ir.SymRef)
-            and new_node.fun.id == "if_"
-            and isinstance(new_node.args[0], ir.Literal)
-        ):  # `if_(True, true_branch, false_branch)` -> `true_branch`
+        if cpm.is_call_to(new_node, ("minimum", "maximum")):
+            # `minimum(a, a)` -> `a`
+            if new_node.args[0] == new_node.args[1]:
+                new_node = new_node.args[0]
+        # `maximum(maximum(__out_size_1, 1), maximum(1, __out_size_1))` -> `maximum(__out_size_1, 1)`
```

Comment:
Let's get rid of this case by canonicalization instead:
```
literal + symref -> symref + literal
```
Then
1. Start
  ```
  maximum(maximum(__out_size_1, 1), maximum(1, __out_size_1))
  ```
2. Canonicalization
  ```
  maximum(maximum(__out_size_1, 1), maximum(__out_size_1, 1))
  ```
3. Simplification `maximum(a, a)` -> `a`:
  ```
  maximum(maximum(1, __out_size_1), maximum(1, __out_size_1))
  ```

The canonicalization also solves the `(literal + (symref + literal))` is not constant folded problem.